app/models/source_set.rb

```diff
@@ -64,4 +65,18 @@ def self.with_tag(tag)
     joins(:tags).where('tags.id = ?', tag.id)
   end
   private_class_method :with_tag
+
+  private
+
+  ##
+  # If the set is being published, save the current timestamp.
+  # If the set was already published, do not save the current timestamp.
+  # If the set is unpublished, clear the timestamp.
+  def published_set
```

Comment:
Could this method name incorporate a verb, like "check publish date" or something similar, since it's not returning a published set, it's doing something to change the set?